ui/src/app/workflows/components/workflows-list/workflows-list.tsx

```diff
@@ -265,31 +266,49 @@ export class WorkflowsList extends BasePage<RouteComponentProps<any>, State> {
                     </div>
-                    {this.state.workflows.map(w => (
-                        <Link
-                            className='row argo-table-list__row'
-                            key={`${w.metadata.namespace}-${w.metadata.name}`}
-                            to={uiUrl(`workflows/${w.metadata.namespace}/${w.metadata.name}`)}>
-                            <div className='columns small-1'>
-                                <PhaseIcon value={w.status.phase} />
-                            </div>
-                            <div className='columns small-3'>{w.metadata.name}</div>
-                            <div className='columns small-2'>{w.metadata.namespace}</div>
-                            <div className='columns small-2'>
-                                <Timestamp date={w.status.startedAt} />
-                            </div>
-                            <div className='columns small-2'>
-                                <Timestamp date={w.status.finishedAt} />
-                            </div>
-                            <div className='columns small-2'>
-                                <Ticker>{() => formatDuration(wfDuration(w.status))}</Ticker>
-                            </div>
-                        </Link>
-                    ))}
+                    {this.state.workflows.map(w => {
+                        return (
+                            <Link
+                                className='row argo-table-list__row'
+                                key={`${w.metadata.namespace}-${w.metadata.name}`}
+                                to={uiUrl(`workflows/${w.metadata.namespace}/${w.metadata.name}`)}>
+                                <div className='columns small-1'>
+                                    <PhaseIcon value={w.status.phase} />
+                                </div>
+                                <div className='columns small-3'>{w.metadata.name}</div>
+                                <div className='columns small-2'>{w.metadata.namespace}</div>
+                                <div className='columns small-1'>
+                                    <Timestamp date={w.status.startedAt} />
+                                </div>
+                                <div className='columns small-1'>
+                                    <Timestamp date={w.status.finishedAt} />
+                                </div>
+                                <div className='columns small-1'>
+                                    <Ticker>{() => formatDuration(wfDuration(w.status))}</Ticker>
+                                </div>
+                                <div className='columns small-3'>
+                                    <div className='workflows-list__labels-container'>
+                                        <WorkflowLabels
+                                            workflow={w}
+                                            onChange={async key => {
```

Comment:
Do we really need `async/await` here? Could you explain why you added it?

Reply:
I had an issue where the state was not set before the `changeFilters` call, since `setState` is asynchronous, so it didn't changing anything. I'll try calling `changeFilters` with the `newTags` variable instead of referencing the state directly and see if that fixes it instead of using an `await`.